File: Ticket_Raising_Backend/TicketClassLibrary/Models/TicketAssignment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TicketClassLibrary.Models;

[Table("TicketAssignment")]
public class TicketAssignment
{
    [Key]
    [Column(TypeName = "char(3)")]
    [Required(ErrorMessage = "Assignment ID is required.")]
    [StringLength(3, MinimumLength = 3, ErrorMessage = "Assignment ID must be exactly 3 characters.")]
    public string AssignmentId { get; set; }

    [ForeignKey(nameof(Ticket))]
    [Column(TypeName = "char(5)")]
    [Required(ErrorMessage = "Ticket ID is required.")]
    [StringLength(5, MinimumLength = 5, ErrorMessage = "Ticket ID must be exactly 5 characters.")]
    public string TicketId { get; set; }

    [ForeignKey(nameof(Employee))]
    [Column(TypeName = "char(5)")]
    [Required(ErrorMessage = "Support Employee ID is required.")]
    [StringLength(5, MinimumLength = 5, ErrorMessage = "Support Employee ID must be exactly 5 characters.")]
    public string Support_Emp_Id { get; set; }

    [Required(ErrorMessage = "Assignment date is required.")]
    public DateTime AssignmentDate { get; set; }

    public virtual Ticket Ticket { get; set; }
    public virtual Employee Employee { get; set; }

    
}

File: Ticket_Raising_Frontend/Ticket_Raising_App/src/Models/TicketAssignment.ts
export class TicketAssignment {
    public assignmentId: string;
    public ticketId: string;
    public Support_Emp_Id: string;
    public assignmentDate: Date;
 
    constructor(
        assignmentId: string = "",
        ticketId: string = "",
        Support_Emp_Id: string = "",
        assignmentDate: Date = new Date()
    ) {
        this.assignmentId = assignmentId;
        this.ticketId = ticketId;
        this.Support_Emp_Id = Support_Emp_Id;
        this.assignmentDate = assignmentDate;
    }
}